pkgs/development/tools/java/dex2jar/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv
+, lib
+, fetchurl
+, jre
+, makeWrapper
+, unzip
+}:
+stdenv.mkDerivation rec {
+  name = "${packageName}-${version}";
+  packageName = "dex2jar";
+  version  = "2.0";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/${packageName}/${name}.zip";
+    sha256 = "1g3mrbyl8sdw1nhp17z23qbfzqpa0w2yxrywgphvd04jdr6yn1vr";
+  };
+
+  nativeBuildInputs = [ makeWrapper unzip ];
+
+
+  unpackPhase = ''
+    unzip $src
+    cd ${name}
+  '';
+
+  phases = [ "unpackPhase" "buildPhase" "installPhase" ];
+
+  buildPhase = ''
+    rm *.bat
+    chmod +x *.sh
+  '';
+
+  installPhase = ''
+    f=$out/lib/dex2jar/
+    bin=$out/bin
+
+    mkdir -p $f $bin
+
+    # fixup path to java
+    sed -i 's#^java#${jre}/bin/java#' d2j_invoke.sh
+
+    mv * $f
+    for i in $f/*.sh; do
+      n=$(basename ''${i%.sh})
+      makeWrapper $i $bin/$n
```

Comment:
```suggestion
      ln -s $i $bin/$n
```

Reply:
An alternative to this is to do a makeWrapper that adds java to the PATH and skip the sed patch thing above.